### PR TITLE
LSO: check availability, update LP (fixes #28936, #28934)

### PR DESCRIPTION
**Pull request by nhaagen**
https://mantis.ilias.de/view.php?id=28936
https://mantis.ilias.de/view.php?id=28934

### CODE REVIEW

File: Modules/LearningSequence/classes/Player/class.ilLSPlayer.php [hunk 1]
```diff
@@ -92,11 +100,15 @@ function ($item) {
...
                 $next_item = $this->getNextItem($current_item, $param);
+                if ($next_item->getAvailability() !== \ILIAS\UI\Component\Listing\Workflow\Step::AVAILABLE) {
+                    $next_item = $current_item;
+                }
                 break;
             case self::LSO_CMD_GOTO:
                 list($position, $next_item) = $this->findItemByRefId($param);
                 break;
-            default: //view-internal / unknown command
+            case self::LSO_CMD_CONTINUE:
```
Comment (klees) [inline Modules/LearningSequence/classes/Player/class.ilLSPlayer.php]: Where does this new command originate from? I do not see that we set this somewhere...

**Review by klees — COMMENTED** [review]
Hi @nhaagen,

frankly, this is hard to understand. I cannot really relate this to the bugs you linked. Please use the extended commit message to elaborate what happens and how this fixes the problem. Is this something we will want to pick to release_6 as well?

Thanks!

File: Modules/LearningSequence/classes/Player/class.ilLSPlayer.php (hunk 1 above)
Comment (nhaagen) [reply Modules/LearningSequence/classes/Player/class.ilLSPlayer.php]: sorry, this was a fragment I forgot to remove.

**Comment by nhaagen** [comment]
Hi @klees ,
I understand, those are tricky ones...
28934:
	retrieve LP immediately, fall into interval afterwards;
	if a LP change occurs BEFORE first javascript-call, the page is not being updated
	for the change is not recognized as such.
	Calling "lso_checkLPOfObject" directly initializes the correct progress at JS-level.

28936:
	If an LSO-Item is reached via the next-command (as it most often is), a page-reload
	will tell the LSO to go to the next item.
	If the item is not available, e.g. by conditions, the current (or last accessible) item should be shown.

Since there are quite general differences between 5.4 and 6 concerning the LSO, I'll check (and fix) seperatedly.

**Pull request merged**